test/test.py

```diff
@@ -1,203 +1,198 @@
-
-    def test_find_all(self):
-        # We should be able to find /all/ the fingerprints that we expect. This
-        # also tests that results given back are equal to the original
-        # fingerprints and not the permuted one.
-        inputs = [1 << i for i in range(63)]
-        self.corpus.insert_bulk(inputs)
-        self.assertEqual(set(self.corpus.find_all((1 << 63))), set(inputs))
-
-    def test_remove(self):
-        # Although I don't imagine there will be much call for this, we should
-        # be able to remove fingerprints from the corpus as well as insert them
-        inputs = [1 << i for i in range(30)]
-        remove = [1 << i for i in range(30, 63)]
-        self.corpus.insert_bulk(inputs)
-        self.corpus.insert_bulk(remove)
-        self.corpus.remove_bulk(remove)
-        self.assertEqual(set(self.corpus.find_all((1 << 63))), set(inputs))
-
-    def test_permute(self):
-        # We should be able to permute and unpermute and always get the
-        # original number back. Unfortunately, it's kind of difficult to
-        # ensure that the permutation is correct, but we'll have a limited
-        # number of examples.
-        import random
-        queries = [random.randint(0, 1 << 63) for i in range(100)]
-
-        # Let's test with a few different combinations of differing bits and
-        # numbers of blocks
-        configs = [(6, 3), (6, 2), (8, 6), (20, 16)]
-        for config in configs:
-            corpus = simhash.Corpus(*config)
-            for t in corpus.tables:
-                for q in queries:
-                    self.assertEqual(t.unpermute(t.permute(q)), q)
-
-    def test_get_all(self):
-        # We should also be able to get a list of all the hashes that are in
-        # the corpus
-        inputs = range(1000)
-        self.corpus.insert_bulk(inputs)
-        self.assertEqual(self.corpus.hashes(), inputs)
-
-if __name__ == '__main__':
-    unittest.main()
+        Who loved an oscilloscope
+          And the cyclical trace
+          Of their carnal embrace
+        Had a damned-near infinite slope!'''
+
+    def compute(self, text):
+        tokens = re.split(r'\W+', text, flags=re.UNICODE)
+        phrases = (' '.join(phrase) for phrase in simhash.shingle(tokens, 4))
+        hashes = map(simhash.unsigned_hash, phrases)
+        return simhash.compute(hashes)
+
+    def test_added_text(self):
+        a = self.compute(self.jabberwocky)
+        b = self.compute(self.jabberwocky + ' - Lewis Carroll (Alice in Wonderland)')
+        self.assertLessEqual(simhash.num_differing_bits(a, b), 3)
+
+    def test_identical_text(self):
+        a = self.compute(self.jabberwocky)
+        b = self.compute(self.jabberwocky)
+        self.assertEqual(0, simhash.num_differing_bits(a, b))
+
+    def test_different(self):
+        a = self.compute(self.jabberwocky)
+        b = self.compute(self.pope)
+        self.assertGreater(simhash.num_differing_bits(a, b), 3)
```

Comment:
Same comment about three bits here. Maybe a constant for `LARGE_BIT_DIFFERENCE` and use it in both places?